### ISSUE TITLE
[NBug] GitExtensions crash after repo removed from disk

### ISSUE THREAD

**Issue by gszo123** [gitextensions/gitextensions#8699]
## Current behaviour

GitExtensions crashes after repo is removed from disk.

## Expected behaviour

No crash.

## Steps to reproduce

1. Open a git repo in GitExtensions
2. Delete the repo from disk
3. GitExtensions crash

## Error Details
```
System.ComponentModel.Win32Exception (0x80004005): Nazwa katalogu jest nieprawidłowa
   w System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
   w System.Diagnostics.Process.Start()
   w GitCommands.Executable.ProcessWrapper..ctor(String fileName, String arguments, String workDir, Boolean createWindow, Boolean redirectInput, Boolean redirectOutput, Encoding outputEncoding)
   w GitCommands.Executable.Start(ArgumentString arguments, Boolean createWindow, Boolean redirectInput, Boolean redirectOutput, Encoding outputEncoding)
   w GitCommands.ExecutableExtensions.<ExecuteAsync>d__11.MoveNext()
--- Koniec śladu stosu z poprzedniej lokalizacji, w której wystąpił wyjątek ---
   w System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   w System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   w Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   w Microsoft.VisualStudio.Threading.JoinableTask`1.CompleteOnCurrentThread()
   w Microsoft.VisualStudio.Threading.JoinableTaskFactory.Run[T](Func`1 asyncMethod, JoinableTaskCreationOptions creationOptions)
   w GitCommands.ExecutableExtensions.Execute(IExecutable executable, ArgumentString arguments, Action`1 writeInput, Encoding outputEncoding, Boolean stripAnsiEscapeCodes)
   w GitCommands.GitModule.GetSelectedBranch(Boolean setDefaultIfEmpty)
   w GitUI.RevisionGridControl.OnGridViewSelectionChanged(Object sender, EventArgs e)
   w System.Windows.Forms.DataGridView.OnSelectionChanged(EventArgs e)
   w System.Windows.Forms.DataGridView.FlushSelectionChanged()
   w System.Windows.Forms.DataGridView.set_NoSelectionChangeCount(Int32 value)
   w System.Windows.Forms.DataGridView.OnCellMouseDown(HitTestInfo hti, Boolean isShiftDown, Boolean isControlDown)
   w System.Windows.Forms.DataGridView.OnCellMouseDown(DataGridViewCellMouseEventArgs e)
   w System.Windows.Forms.DataGridView.OnMouseDown(MouseEventArgs e)
   w GitUI.UserControls.RevisionGrid.RevisionDataGridView.OnMouseDown(MouseEventArgs e)
   w System.Windows.Forms.Control.WmMouseDown(Message& m, MouseButtons button, Int32 clicks)
   w System.Windows.Forms.Control.WndProc(Message& m)
   w System.Windows.Forms.DataGridView.WndProc(Message& m)
   w System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   w System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   w System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

## Additional information
Git repository removed from disk before closing GitExtensions

## Environment
- Git Extensions 3.4.3.9999
- Build d4b0f48bbf3e39a71f5d7ff5231be5678d4aa0f1
- Git 2.29.2.windows.2
- Microsoft Windows NT 10.0.19042.0
- .NET Framework 4.8.4300.0
- DPI 96dpi (no scaling)

**Comment by gerhardol** [comment]
New issue but no exception after #8534 

But if the repo is removed an exception is not so strange, not a prio to handle...
